src/ripple/consensus/Validations.h

```diff
@@ -0,0 +1,504 @@
+
+            // Sequence of 0 indicates a missing sequence number
+            if (oldSeq && newSeq && oldSeq == newSeq)
+            {
+                result = AddOutcome::sameSeq;
+
+                // Remove current validation from the ledger set
+                // for the revoked signing key
+                if (val.key() != oldVal.key())
+                {
+                    auto set = byLedger_.find(oldVal.ledgerID());
+                    if (set != byLedger_.end())
+                    {
+                        set->second.erase(key);
+                        // Erase the set if it is now empty
+                        if (set->second.empty())
+                            byLedger_.erase(set);
+                    }
+                }
+            }
+
+            if (val.signTime() > oldVal.signTime() || val.key() != oldVal.key())
+            {
+                // This is either a newer validation or a new signing key
+                auto const oldID = oldVal.ledgerID();
+                // Allow impl to take over oldVal
+                impl().onStale(std::move(oldVal));
+                // Replace old val in the map and set the previous ledger ID
+                ins.first->second = val;
+                ins.first->second.setPreviousLedgerID(oldID);
+            }
+            else
+            {
+                // We already have a newer validation from this source
+                result = AddOutcome::stale;
+            }
+        }
+
+        return result;
+    }
+
+    /** Expire old validation sets
+
+        Remove validation sets that were accessed more than
+        VALIDATION_SET_EXPIRES ago.
+    */
+    void
+    expire()
+    {
+        beast::expire(byLedger_, parms_.VALIDATION_SET_EXPIRES);
+    }
+
+    struct ValidationCounts
+    {
+        //! The number of trusted validations
+        std::size_t count;
+        //! The highest trusting node ID
+        NodeID highNode;
+    };
+
+    /** Distribution of current trusted validations
+
+        Calculates the distribution of current validations but allows
+        ledgers one away from the current ledger to count as the current.
+
+        @param t Current time used to determine ledger staleness
+        @param currentLedger The identifier of the ledger we believe is current
+        @param priorLedger The identifier of our previous current ledger
+        @param cutoffBefore Ignore ledgers with sequence number before this
+        @param j Journal for logging
+    */
+    hash_map<LedgerID, ValidationCounts>
+    currentTrustedDistribution(
+        NetClock::time_point t,
+        LedgerID const& currentLedger,
+        LedgerID const& priorLedger,
+        std::size_t cutoffBefore,
+        beast::Journal& j)
+    {
+        bool valCurrentLedger = currentLedger != beast::zero;
+        bool valPriorLedger = priorLedger != beast::zero;
+
+        hash_map<LedgerID, ValidationCounts> ret;
+
+        current(t, [&](auto const&, auto const& v) {
```

Comment:
I'm not advocating for a rule.  However I'd like to point out another example in the code base that I was just struggling with:

https://github.com/ripple/rippled/blob/develop/src/ripple/rpc/impl/ServerHandlerImp.cpp#L363-L365

The declaration looks like this:
```
    m_jobQueue.postCoro(jtCLIENT, "WS-Client",
        [this, session = std::move(session),
            jv = std::move(jv)](auto const& c)
        {
        ...
```
My question to you is this: as a maintainer of this code, what is the type of `c`?  [Mickey doesn't get to answer 😄 ]?

It turns out that 'c' can *only* be one type: `std::shared_ptr<JobQueue::Coro> const&`.

The length of that type name might very well be past the threshold of someone looking for a terse syntax.  But, for someone reading the code, the information is difficult to extract by just looking at the sources.  I guess my point is that a long type name can look like line noise unless you need to know what the type is.  Then it changes from line noise into valuable information.

Again, this is just my personal take on the subject.  I'm not arguing for a rule or for a change to the coding guidelines.  Thanks for listening.